app/controllers/marks_graders_controller.rb

```diff
@@ -60,24 +60,21 @@ def download_grader_students_mapping
     grade_entry_form = GradeEntryForm.find(params[:grade_entry_form_id])
     students = students_with_assoc
 
-    file_out = CSV.generate do |csv|
-      students.each do |student|
-        # csv format is student_name, ta1_name, ta2_name, ... etc
-        student_array = [student.user_name]
-        grade_entry_student = student.grade_entry_students.find do |entry|
-          entry.grade_entry_form_id == grade_entry_form.id
-        end
-        unless grade_entry_student.nil?
-          grade_entry_student.tas.order(:user_name).each do |ta|
-            student_array.push(ta.user_name)
-          end
+    file_out = MarkusCSV.generate(students) do |student|
+      # csv format is student_name, ta1_name, ta2_name, ... etc
+      student_array = [student.user_name]
+      grade_entry_student = student.grade_entry_students.find do |entry|
+        entry.grade_entry_form_id == grade_entry_form.id
+      end
+      unless grade_entry_student.nil?
+        grade_entry_student.tas.order(:user_name).each do |ta|
```

Comment:
1. It's cleaner to do `student_array.concat(...)`, if possible.
2. Since you're only using usernames, you should be able to use `pluck` here effectively.